src/wallet/api/wallet_manager.cpp

```diff
@@ -57,9 +57,12 @@ Wallet *WalletManagerImpl::openWallet(const std::string &path, const std::string
     return wallet;
 }
 
-Wallet *WalletManagerImpl::recoveryWallet(const std::string &path, const std::string &memo, bool testnet)
+Wallet *WalletManagerImpl::recoveryWallet(const std::string &path, const std::string &memo, bool testnet, uint64_t restoreHeight)
 {
     WalletImpl * wallet = new WalletImpl(testnet);
+    if(restoreHeight > 0){
```

Comment:
That would require the daemon to be running. Since that's not always the case I would suggest having that type of check when the actual sync occurs. I put up an issue yesterday and someone already seems to be working on it =)
https://github.com/monero-project/monero/issues/1190